### PR TITLE
Add a HTTP server to packet.net to serve iPXE files

### PR DESCRIPTION
**Pull request by rn**
This adds a `-serve` option to the packet.net backend to serve the iPXE files locally. In particualr with `ngrok` this makes booting locally build images via iPXE much simpler. The server only serves the files needed by iPXE. See updated docs for an example

image

### CODE REVIEW

File: src/cmd/linuxkit/run_packet.go [hunk 1]
```diff
@@ -267,3 +302,28 @@ func sshHostKey(host string) (ssh.PublicKey, error) {
...
 	return hostKey, nil
 }
+
+// This implements a http.FileSystem which only responds to specific files.
+type serveFiles struct {
+	files []string
+}
+
+func (fs serveFiles) Open(name string) (http.File, error) {
+	name = strings.TrimLeft(name, "/")
```
Comment (justincormack) [inline src/cmd/linuxkit/run_packet.go]: shouldnt this account for Windows file seperator?

Comment (rn) [inline src/cmd/linuxkit/run_packet.go]: not this part, but some other part, I think. URL paths are always `/` but the filename passed in might be windows so the `name == fn` may fail

Comment (justincormack) [inline src/cmd/linuxkit/run_packet.go]: Are you going to make changes then?

Comment (rn) [inline src/cmd/linuxkit/run_packet.go]: yes, but then started adding other bits while at it...

**Comment by rn** [comment]
Update pushed. This now uses the official packet go bindings as they incorporated the iPXE changes and, by default removes the machine when done

**Pull request merged**